### PR TITLE
documentation improvement

### PR DESCRIPTION
**Pull request by Titotix**
Related to previously issue opened #102 

- mention code examples in documentation
- mention `userinfo_endpoint` key on `ProviderMetadata`
- mention `OIDCAuthentication.init_app()`
- add docstring on `ProviderMetadata` and `ClientMetadata`

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -25,13 +25,17 @@ from flask_pyoidc.provider_configuration import ProviderConfiguration, ProviderM
...
+                                     jwks_uri='https://op.example.com/jwks',
+                                     userinfo_endpoint='https://op.example.com/userinfo')
 config = ProviderConfiguration(provider_metadata=provider_metadata, [client configuration])
 ```
 
 See the OpenID Connect specification for more information about the
 [provider metadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).
 
+As mentionned in OpenID specification, `userinfo_endpoint` is optionnal. If it's
+not provided, no userinfo request will be done. `flask_pyoidc.UserSession.userinfo` will be set to `None`.
```
Comment (zamzterz) [inline docs/configuration.md:37]: Some small typos:
* "As mention~n~ed in OpenID _Connect_ specification"
* "optionnal" -> "optional"

Suggestion:
* "done. `flask_pyoidc.UserSession.userinfo` [...]" -> "done and`flask_pyoidc.UserSession.userinfo` [...]"

File: docs/quickstart.md [hunk 2]
```diff
@@ -26,12 +26,31 @@ def index():
...
+    return app
+```
+
 After a successful login, this extension will place three things in the user session (if they are received from the
 provider):
 * [ID Token](http://openid.net/specs/openid-connect-core-1_0.html#IDToken)
 * [Access Token](http://openid.net/specs/openid-connect-core-1_0.html#TokenResponse)
 * [Userinfo Response](http://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse)
 
+On the side of this documentation, you may have a look on some
```
Comment (zamzterz) [inline docs/quickstart.md:51]: "on some" -> "on a"

File: docs/quickstart.md [hunk 3]
```diff
@@ -26,12 +26,31 @@ def index():
                    userinfo=user_session.userinfo)
 ```
 
+You can also use the Flask's application factory
```
Comment (zamzterz) [inline docs/quickstart.md:29]: "use the Flask's application factory" -> "use a Flask application factory"

File: docs/index.rst [hunk 4]
```diff
@@ -1,6 +1,11 @@
 Welcome to flask-pyoidc's documentation!
 ========================================
 
+On the side of this documentation, you may have a look on some
```
Comment (zamzterz) [inline docs/index.rst:4]: Suggestion: "On the side of [...]" -> "In addition to [...]"

File: src/flask_pyoidc/provider_configuration.py [hunk 5]
```diff
@@ -58,6 +58,17 @@ def to_dict(self):
 
 class ProviderMetadata(OIDCData):
     def __init__(self, issuer=None, authorization_endpoint=None, jwks_uri=None, **kwargs):
+        """
+        Args:
```
Comment (zamzterz) [inline src/flask_pyoidc/provider_configuration.py:62]: This is great! 👍

**Review by zamzterz — CHANGES_REQUESTED** [review]
This is great documentation additions, thank you! 👏 

I've added some small comments of things to be changed.

File: docs/configuration.md [hunk 6]
```diff
@@ -25,13 +25,17 @@ from flask_pyoidc.provider_configuration import ProviderConfiguration, ProviderM
...
-                                     jwks_uri='https://op.example.com/jwks')
+                                     jwks_uri='https://op.example.com/jwks',
+                                     userinfo_endpoint='https://op.example.com/userinfo')
 config = ProviderConfiguration(provider_metadata=provider_metadata, [client configuration])
 ```
 
 See the OpenID Connect specification for more information about the
 [provider metadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).
 
+As mentionned in OpenID Connect specification, `userinfo_endpoint` is optional. If it's
```
Comment (zamzterz) [inline docs/configuration.md:36]: Small typo: "mentionned" -> "mentioned"

**Review by zamzterz — CHANGES_REQUESTED** [review]
Just one more tiny thing, then it'll be ready for merging! 🎉

File: docs/configuration.md (hunk 6 above)
Comment (Titotix) [reply docs/configuration.md]: ahah, yes, I fixed  it, french habit to put a second consonant before a vowel

**Review by zamzterz — APPROVED** [review]
Perfect 💯 

Thanks again for the great contribution! 🙇

**Pull request merged**